File: api-demo.go
package main

import (
	"context"
	"fmt"
	"log"

	"google.golang.org/grpc"
	"v2ray.com/core/app/proxyman/command"
	"v2ray.com/core/common/protocol"
	"v2ray.com/core/common/serial"
	"v2ray.com/core/proxy/vmess"
)

const (
	API_ADDRESS = "127.0.0.1"
	API_PORT    = 10085
	INBOUND_TAG = "proxy"
	LEVEL       = 0
	EMAIL       = "123@gmail.com"
	UUID        = "2601070b-ab53-4352-a290-1d44414581ee"
	ALTERID     = 32
)

func addUser(c command.HandlerServiceClient) {
	resp, err := c.AlterInbound(context.Background(), &command.AlterInboundRequest{
		Tag: INBOUND_TAG,
		Operation: serial.ToTypedMessage(&command.AddUserOperation{
			User: &protocol.User{
				Level: LEVEL,
				Email: EMAIL,
				Account: serial.ToTypedMessage(&vmess.Account{
					Id:               UUID,
					AlterId:          ALTERID,
					SecuritySettings: &protocol.SecurityConfig{Type: protocol.SecurityType_AUTO},
				}),
			},
		}),
	})
	if err != nil {
		log.Printf("failed to call grpc command: %v", err)
	} else {
		log.Printf("ok: %v", resp)
	}
}
func removeUser(c command.HandlerServiceClient) {
	resp, err := c.AlterInbound(context.Background(), &command.AlterInboundRequest{
		Tag: INBOUND_TAG,
		Operation: serial.ToTypedMessage(&command.RemoveUserOperation{
			Email: EMAIL,
		}),
	})
	if err != nil {
		log.Printf("failed to call grpc command: %v", err)
	} else {
		log.Printf("ok: %v", resp)
	}
}
func main() {
	cmdConn, err := grpc.Dial(fmt.Sprintf("%s:%d", API_ADDRESS, API_PORT), grpc.WithInsecure())
	if err != nil {
		panic(err)
	}
	hsClient := command.NewHandlerServiceClient(cmdConn)
	addUser(hsClient)
	// removeUser(hsClient)
}

File: api-demo.js
var grpc = require('grpc')
var services = require('./v2ray.com/core/app/proxyman/command/command_grpc_pb')
var command = require('./v2ray.com/core/app/proxyman/command/command_pb')
var protocol_user = require('./v2ray.com/core/common/protocol/user_pb')
var protocol_headers = require('./v2ray.com/core/common/protocol/headers_pb')
var serial = require('./v2ray.com/core/common/serial/typed_message_pb')
var vmess = require('./v2ray.com/core/proxy/vmess/account_pb')

var hsClient = new services.HandlerServiceClient('127.0.0.1:1088', grpc.credentials.createInsecure())


var securitySettings = new protocol_headers.SecurityConfig()
securitySettings.setType(protocol_headers.SecurityType.AUTO)

var v2Account = new vmess.Account()
v2Account.setId('d6f5e396-cccc-4e45-b0eb-a953105cc262')
v2Account.setAlterId(32)
v2Account.setSecuritySettings(securitySettings)
var v2AccountTypedMessage = new serial.TypedMessage()
v2AccountTypedMessage.setType('v2ray.core.proxy.vmess.Account')
v2AccountTypedMessage.setValue(v2Account.serializeBinary())

var v2User = new protocol_user.User()
v2User.setLevel(1)
v2User.setEmail('tims@rayku.net')
v2User.setAccount(v2AccountTypedMessage)

var v2AddUserOperation = new command.AddUserOperation()
v2AddUserOperation.setUser(v2User)
var v2AddUserOperationTypedMessage = new serial.TypedMessage()
v2AddUserOperationTypedMessage.setType('v2ray.core.app.proxyman.command.AddUserOperation')
v2AddUserOperationTypedMessage.setValue(v2AddUserOperation.serializeBinary())

var v2AlterInboundRequest = new command.AlterInboundRequest()
v2AlterInboundRequest.setTag('tcp')
v2AlterInboundRequest.setOperation(v2AddUserOperationTypedMessage)

hsClient.alterInbound(v2AlterInboundRequest, function(err, resp){
    if (err) {
        if (err.code==2){
            console.log('user already exists')
        } else {
            console.log(err)
        }
        return
    }
    console.log('ok')
})